### ISSUE TITLE
rpm signing fails with gpg-agent 2.1.9

### ISSUE THREAD

**Issue by danielmitterdorfer** [elastic/elasticsearch#17053]
**Elasticsearch version**: 2.x branch

**OS version**: Fedora 23

**Description of the problem including expected versus actual behavior**:

Currently our [Fedora-based build jobs for 2.x](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+2.x+multijob-os-compatibility/os=fedora/) fail with the following message:

```
[INFO] spawn rpm --define _gpg_name 16E55242 --define _gpg_path /var/lib/jenkins/workspace/elastic+elasticsearch+2.x+multijob-os-compatibility/os/fedora/distribution/src/test/resources/dummyGpg --addsign elasticsearch-2.3.0-SNAPSHOT20160309223154.noarch.rpm
[INFO] elasticsearch-2.3.0-SNAPSHOT20160309223154.noarch.rpm:
[INFO] gpg: WARNING: unsafe permissions on homedir '/var/lib/jenkins/workspace/elastic+elasticsearch+2.x+multijob-os-compatibility/os/fedora/distribution/src/test/resources/dummyGpg'
[INFO] gpg: starting migration from earlier GnuPG versions
[INFO] gpg: can't connect to the agent: Invalid value passed to IPC
[INFO] gpg: error: GnuPG agent unusable. Please check that a GnuPG agent can be started.
[INFO] gpg: migration aborted
[INFO] gpg: can't connect to the agent: Invalid value passed to IPC
[INFO] gpg: skipped "16E55242": No secret key
[INFO] gpg: signing failed: No secret key
Build was aborted
Finished: ABORTED
```

The root cause that gpg-agent (installed version 2.1.9) creates a Unix socket with the following path name which is too long (at most 108 bytes are allowed):

```
[jenkins@slave-2f48afe8 dummyGpg]$ gpg-agent --homedir . --daemon -vvv
gpg-agent[28440]: socket name '/var/lib/jenkins/workspace/elastic+elasticsearch+2.x+multijob-os-compatibility/os/fedora/distribution/src/test/resources/dummyGpg/S.gpg-agent' is too long
```

**Steps to reproduce**:
 1. Run `mvn verify` on Fedora 23

**Comment by danielmitterdorfer** [comment]
A potential workaround for this issue is to add a symlink to a shorter path (e.g. some subdirectory in the tmp folder) and use that instead of the original path. The [GnuPG team is already discussing on shortening the Unix socket path](http://comments.gmane.org/gmane.comp.encryption.gpg.devel/21073) in a future version.

**Comment by nik9000** [comment]
Is there some argument we can give that'll stop trying to use the agent and just use the nasty expect hacks that every other OS uses?

**Comment by danielmitterdorfer** [comment]
I am not really familiar how the process works but the maven-rpm-plugin provides the [parameters](http://www.mojohaus.org/rpm-maven-plugin/rpm-mojo.html) `keyPassphrase`, `keyname` and `keypath` [which we already set](https://github.com/elastic/elasticsearch/blob/2.x/distribution/rpm/pom.xml#L109-L113). Also, the maven-rpm-plugin [creates an expect script](https://github.com/mojohaus/rpm-maven-plugin/blob/0efd27d3beb64c291b960731fe1bdbbb5dbd938e/src/main/java/org/codehaus/mojo/rpm/RPMSigner.java#L137).

**Comment by danielmitterdorfer** [comment]
@nik9000: Another "solution" to this problem is to deactivate signing on this specific platform with the `skipSign` property that you've introduced recently. I am not sure whether we really want to test dummy-signing on this platform. Especially considering that the symlink workaround is also just a workaround IMHO. Wdyt?

**Comment by danielmitterdorfer** [comment]
I've just learned that we cannot deactivate signing on a specific platform in the current build infrastructure, leaving us with the symlink workaround for the Maven build. :(

**Comment by dliappis** [comment]
re: https://github.com/elastic/elasticsearch/issues/17053#issuecomment-195303351 there could be some refactoring in the jjb jobs but it would essentially break the logic of one job definition for all platforms which is one important part of the simplication/devops style management of the new CI system.

However, cc'ing @elasticdog directly here in case he has some other idea.

**Comment by dliappis** [comment]
I looked into bypassing what exactly rpm invokes when signing with gpg and found a rather quite informative answer in the fedora forums.

In [that answer](https://ask.fedoraproject.org/en/question/56107/can-gpg-agent-be-used-when-signing-rpm-packages/?answer=57111#post-id-57111) the actual gpg sign command is explicitly defined, so as suggested by @nik9000 then I looked at a way to skip the gpg-agent. Unfortunately with gpg2 this is not possible, as `man gpg2` says:

```
      --use-agent

       --no-use-agent
              This is dummy option. gpg2 always requires the agent.
```

Still looking at other hacks we could put in the definition of `__gpg_sign_cmd`

**Comment by jasontedor** [comment]
I think this should be closed since #17070 and #17073 were integrated? I think that this was not closed automatically because they were not merged into master, but instead into 2.x. Please reopen if this I'm closing this in error.

**Comment by danielmitterdorfer** [comment]
@jasontedor You were right; this was just an oversight. Thanks for closing.
